src/Adapter/MSTest.TestAdapter/MSTestSettings.cs

```diff
@@ -245,6 +245,7 @@ public static void PopulateSettings(MSTestSettings settings)
     /// <param name="context">
     /// The discovery context that contains the runsettings.
     /// </param>
+    [Obsolete("this function will be removed in v4.0.0")]
     public static void PopulateSettings(IDiscoveryContext? context)
```

Comment:
You could reduce duplicate code by having this method calling `PopulateSettings(context ,null)`